src/activity_handlers.cpp

```diff
@@ -1964,7 +1987,12 @@ void activity_handlers::start_fire_do_turn( player_activity *act, Character *you
     }
 
     item &firestarter = *act->targets.front();
-    if( firestarter.has_flag( flag_REQUIRES_TINDER ) ) {
+
+    const furn_id f_id = here.furn( here.get_bub( act->placement ) );
+    const bool is_smoker = f_id == furn_f_smoking_rack ||
+                           f_id == furn_f_metal_smoking_rack;
+
+    if( firestarter.has_flag( flag_REQUIRES_TINDER ) && !is_smoker ) {
```

Comment:
this doesn't check for kiln

Reply:
Correct. That's because I _am_ actually able to put tinder inside of the kiln, just not inside of the smoking rack. Do you still want me to add it?

Reply:
No, this is good!